mlflow/transformers/model_io.py

```diff
@@ -123,7 +128,40 @@ def _load_component(flavor_conf, name, local_path=None):
         return cls.from_pretrained(repo, revision=revision)
 
 
-def _load_model(model_name_or_path, flavor_conf, accelerate_conf, device, revision=None):
+# FOR DEMO ONLY.
+#
+# inspect the HF config file to find the correct path to load the model from.
+#
+# this only works if we save the model's remote code in the `code` directory,
+# but should be pretty doable. for example, we can check if the model relies on
+# external code by doing something like:
+#
+# `transformers.utils.HF_MODULES_CACHE in inspect.getfile(model.__class__)`
+#
+# the HF config file contains the mapping between normal transformers constructs
+# and the custom code that was defined, so it should be as close to a source
+# of truth as we can get. internally, transformers does something similar,
+# see `get_class_from_dynamic_module()`:
+#
+# https://github.com/huggingface/transformers/blob/b6404866cda2952942a39e424206aed0cd6aeb5c/src/transformers/models/auto/auto_factory.py#L428
+#
+# there are some issues to work out with regard to namespace collision, etc.
```

Comment:
Are there similar considerations addressed within their cached synthetic __init__ that they create within their library for consuming the auto_map mapping?

Reply:
Yes, they prefix the modules with the revision hash that the files are from, which I suppose would virtually eliminate collision concerns. However, there doesn't seem to be a good way to retrieve this given a transformers model. I explored some options like inspecting the file that the class is defined in and then traversing up to the HF cache directory to get the revision, but it seemed super hacky.

Also, I'm not sure if we would have a similar construct for custom models not on HF (though of course we could scope down this fix to require people to push to HF hub for it to work).